web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/GetCertificateTypesFacadeServiceImpl.java

```diff
@@ -93,6 +96,19 @@ private CertificateTypeInfoDTO addAdditionalResourceLinks(CertificateTypeInfoDTO
         return intygModule;
     }
 
+    private CertificateTypeInfoDTO addConfirmationModal(CertificateTypeInfoDTO intygModule, Personnummer patientId) {
+        final var provider = ConfirmationModalProviderResolver.get(intygModule.getId(), CertificateStatus.UNSIGNED, "NORMAL", true);
+        final var latestCertificateVersion = intygTextsService.getLatestVersion(intygModule.getId());
+        final var patient = patientDetailsResolver.resolvePatient(patientId, intygModule.getId(), latestCertificateVersion);
+        if (provider != null) {
```

Comment:
Should we move this if-statement so that its done right after ```final var provider = ConfirmationModalProviderResolver.get(intygModule.getId(), CertificateStatus.UNSIGNED, "NORMAL", true);```?

Should not affect performance, but seems uneccessary to ask PU for patient information if we dont need it.

Something like this:

```java
final var provider = ConfirmationModalProviderResolver.get(intygModule.getId(), CertificateStatus.UNSIGNED, "NORMAL", true);
if (provider != null) {
     final var latestCertificateVersion = intygTextsService.getLatestVersion(intygModule.getId());
     final var patient = patientDetailsResolver.resolvePatient(patientId, intygModule.getId(), latestCertificateVersion);
     intygModule.setConfirmationModal(
                provider.create(patient.getFullstandigtNamn(), patientId.getPersonnummerWithDash(), "NORMAL")
            );
}

return intygModule;
```

Reply:
Sure